activation/activation_test.go

```diff
@@ -314,6 +314,15 @@ func TestBuilder_RestartSmeshing(t *testing.T) {
 	})
 }
 
+// This test is intended to check that if PrepareInitializer returns an error
+// then that error is returned instead of crashing, which is what should happen if
+// StartSession returns an error.
```

Comment:
i think such comments are not useful, the code below expresses very clearly what is the test about

Reply:
The reason I added this test is to check that we don't crash when StartSmeshing is called and PrepareInitializer returns an error. There's nothing in the code of this test that would allow you to understand that.

Ideally I'd test for a panic when StartSession is called and then verify that if PrepareInitializer returns an error there is no panic, this would then make it clear from the code that PrepareInitializer returning an error ensures that no panic occurs. But we can't test for a panic because StartSession starts its own go-routine to run StartSession.

Reply:
I'm leaning towards @dshulyak opinion.

We don't want to test everything for "does not panic", here we had a bug/regression that caused a panic in a go routine started by `StartSession`. Ideally we have a test that would have failed without the fix and now passes.

Before it didn't panic it was failing silently. So imo a proper test should return a non-`nil` error on `StartSmeshing` with gomock and assert that `log.Panic` is called (while before it would silently return).

This test is additional to the new behavior to check if when `PrepareInit` returns an error that this error is returned by `StartSession` without calling `StartSmeshing`.

I prefer good names for those 2 tests (e.g. `Test_PanicOnErrStartSmeshing` and `Test_ErrOnPrepareInitHandeled` (or similar) and if additional context is necessary just add a comment with the link to the issue that the test belongs to (`// regression test for https://github.com/spacemeshos/go-spacemesh/issues/4292`).

Reply:
Ok, so the thing I'm missing is how to assert that  `log.Panic` is called?

Reply:
by recovering from panic, or there are some nuances? i think there is some helper like require.Panic

Reply:
We have a `log.Logger` interface. `NewBuilder` can take this a parameter instead of the concrete type `log.Log`. Then you an mock the logger for tests. 🙂 

`log.Logger` needs to be extended to include a `Panic` method for this to work, but the change is small enough that imo it can be part of this PR.

Reply:
@dshulyak the nuance is that recover only works to recover from panics emitted by the same goroutine that recover is called in and in this case the builder starts a new goroutine to run `StartSession` and so we cannot recover panics from that.

Reply:
@fasmat, I will try mocking the log